hyperspy/signal.py

```diff
@@ -1924,8 +1924,18 @@ def _get_undefined_axes_list(self):
     def __call__(self, axes_manager=None):
         if axes_manager is None:
             axes_manager = self.axes_manager
-        return np.atleast_1d(
-            self.data.__getitem__(axes_manager._getitem_tuple))
+        if getattr(self._plot, 'pointer', None) is not None:
+            indices = self._plot.pointer.indices
+            size = self._plot.pointer.get_size_in_indices()
+            array_slices = [slice(None, None, None)] * \
+                self.axes_manager.navigation_dimension
+            for i, axis in enumerate(self.axes_manager.navigation_axes[::-1]):
```

Comment:
I'm not sure what you mean by the `width`. The width of the full axis? Or just the width of the pointer that is currently slicing?

Now that I think about it, I quite like the idea of storing (instead of just the indices of the current "position") spans/slices of the selection independently for each axis. This would allow visually interactively slicing 3-4-5-nD navigation spaces. For example, plot as `<1,2| 3,4,5>`, set the regions of interest for (1,2), then plot as `<2,3|1,4,5>` - the width from (2) is still preserved, and you can adjust (3,) to match whatever you want, and then continue onwards. I quite like such proposal.. 

The immediate drawback seems to be not slicing too many axes when recreating the default behavior, but that will have to be resolved with sane defaults.

@francisco-dlp what do you think?